server/handler/details.go

```diff
@@ -121,31 +120,33 @@ func (h *Details) ServeHTTP(w http.ResponseWriter, r *http.Request) error {
 
 	config, err := h.ConfigFetcher.ConfigForPR(ctx, prctx, client)
 	data.PolicyURL = getPolicyURL(pr, config)
-
 	if err != nil {
-		data.Error = errors.WithMessage(err, fmt.Sprintf("Failed to fetch configuration at ref=%s", config.Ref))
+		data.Error = errors.WithMessage(err, fmt.Sprintf("Failed to fetch configuration: %s", config))
 		return h.render(w, data)
 	}
 
-	if config.Missing() {
-		data.Error = errors.New(config.Description())
-		return h.render(w, data)
-	}
+	evaluator, err := h.Base.ValidateFetchedConfig(ctx, prctx, client, config, common.TriggerAll)
 
-	if config.Invalid() {
-		data.Error = errors.WithMessage(config.Error, config.Description())
+	if err != nil {
+		data.Error = err
 		return h.render(w, data)
 	}
 
-	evaluator, err := policy.ParsePolicy(config.Config)
-	if err != nil {
-		data.Error = errors.WithMessage(err, fmt.Sprintf("invalid policy at ref \"%s\"", config.Ref))
+	if evaluator == nil {
+		data.Error = errors.Errorf("Unable to evaluate: %s", config.Description())
 		return h.render(w, data)
 	}
 
-	result := evaluator.Evaluate(ctx, prctx)
+	result, err := h.Base.EvaluateFetchedConfig(ctx, prctx, client, evaluator, config)
 	data.Result = &result
 
+	if err != nil {
+		if _, ok := err.(*pull.TemporaryError); ok {
+			err = errors.WithMessage(err, "This error may be temporary. Wait 30 seconds and refresh this page to retry")
+		}
```

Comment:
This will read in the UI as something like:

```
This error may be temporary. Wait 30 seconds and refresh this page to retry: failed to do a thing: unable to get the stuff: unexpected status doing my job.
```

While a bit more work, I think it would be nice to add an `IsTemporaryError` field to `data` and then add this help message as part of the UI rendering. That way, it could appear after the error, which I think is more natural.

This also raises a good related point about error wrapping. The message is meant to be additional context that appears as a prefix of the message already in the error. These can get pretty long so it's worth thinking about when additional context is actually valuable and then phrasing that context to succinct and direct.

It's unfortunately pretty common to end up with errors that read like "failed to X: unable to X: could not X: the actual reason X failed", where the context doesn't really add much.

Reply:
Sounds good, added it as an element in the details template.